# Notebook 01 — Data Loading and Overview

# Customer Satisfaction Analysis in Brazilian E-Commerce

## Project Introduction
I selected the Brazilian E-Commerce dataset because it is a real 
world business dataset that closely mirrors the kind of data 
encountered in production environments. It is particularly strong 
for understanding customer behavior, building reliable data 
pipelines, and converting raw data into decision-ready outputs. 
The dataset contains real company data including orders, payments, 
customer reviews, and delivery information, which allowed me to 
demonstrate end-to-end data handling, exploratory analysis, and 
predictive modeling skills.

## Business Question
What factors most influence customer satisfaction in e-commerce, 
and can we predict whether a customer will leave a positive or 
negative review based on their order experience?



## Setup
Libraries and data are loaded using the shared utility module to keep the notebook clean and consistent.

In [6]:
import sys
sys.path.append("..")
from utils import *

df = load_data()
print("Data loaded successfully:", df.shape)

Data loaded successfully: (119143, 28)


## Data Loading and Merging
The 5 datasets are loaded into separate dataframes and merged 
together into one master dataframe using the order_id and customer_id 
as the joining keys. This gives us a complete view of each order 
including the customer review score, payment details, and delivery 
information all in one place.

In [7]:
# Load all datasets
import sys
sys.path.append("..")
from utils import load_data

df = load_data()
print("Data loaded successfully:", df.shape, df.head())

Data loaded successfully: (119143, 28)                            order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
4  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
3    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
4    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1        

### Data Loading  Result
The five datasets were successfully merged into one master dataframe containing 119,143 rows and 28 columns. Each row represents a unique order with its associated review score, payment details, and customer information all in one place.

## Data Overview
In this section I take a first look at the master dataframe to understand its structure, size, data types, and identify any missing values before moving into the analysis. The results will inform the cleaning approach used in the next notebook.

In [5]:
print("Shape:", df.shape) # to check the shape 

print("\nData Type:")
print(df.dtypes) # to see the datatype of each column

print("\nMissing Values:")
print(df.isnull().sum()) #for missing values

print("\nStatistical Summary:")
print (df.describe()) # To statistivally summaries the data. mean mode median standard deviation 

Shape: (119143, 28)

Data Type:
order_id                          object
customer_id                       object
order_status                      object
order_purchase_timestamp          object
order_approved_at                 object
order_delivered_carrier_date      object
order_delivered_customer_date     object
order_estimated_delivery_date     object
review_id                         object
review_score                     float64
review_comment_title              object
review_comment_message            object
review_creation_date              object
review_answer_timestamp           object
order_item_id                    float64
product_id                        object
seller_id                         object
shipping_limit_date               object
price                            float64
freight_value                    float64
payment_sequential               float64
payment_type                      object
payment_installments             float64
payment_value            

## Data overview findings
The master dataset contains 119,143 rows and 28 columns. A review of data types, missing values, and summary statistics shows the following:
- Several date fields are stored as strings and will require conversion to proper datetime format.
- The review_score column is the target variable, ranging from 1 to 5, with an average score of 4.0.
- review_comment_title and review_comment_message contain substantial missing values and will be removed due to limited usefulness.
- order_delivered_customer_date has 3,421 missing entries, likely representing undelivered orders.
- review_id and related review fields have 997 missing values, indicating orders without submitted reviews.
- Payment related columns contain only 3 missing values, which is negligible.